cfgmgr/vrfmgr.cpp

```diff
@@ -0,0 +1,194 @@
+#include <string.h>
+#include "logger.h"
+#include "dbconnector.h"
+#include "producerstatetable.h"
+#include "tokenize.h"
+#include "ipprefix.h"
+#include "vrfmgr.h"
+#include "exec.h"
+#include "shellcmd.h"
+
+#define VRF_TABLE_START 1001
+#define VRF_TABLE_END 2000
+
+using namespace swss;
+
+VrfMgr::VrfMgr(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, const vector<string> &tableNames) :
+        Orch(cfgDb, tableNames)
+{
+    for (uint32_t i = VRF_TABLE_START; i < VRF_TABLE_END; i++)
+    {
+        m_freeTables.emplace(i);
+    }
+
+    /* Get existing VRFs from Linux */
+    stringstream cmd;
+    string res;
+
+    cmd << IP_CMD << " -d link show type vrf";
+    int ret = swss::exec(cmd.str(), res);
+    if (ret)
+    {
+        SWSS_LOG_ERROR("Command %s failed, rc: %i", res.c_str(), ret);
```

Comment:
Probably EXEC_WITH_ERROR_THROW(cmd, res) is better here.

Reply:
Done